File: packages/app-check/ios/Plugin/FirebaseAppCheck.swift
import Foundation

import FirebaseCore
import FirebaseAppCheck

@objc public class FirebaseAppCheck: NSObject {

    override public init() {
        if FirebaseApp.app() == nil {
            FirebaseApp.configure()
        }
    }

    @objc public func getToken(forceRefresh: Bool, completion: @escaping (String?, Int, Error?) -> Void) {
        AppCheck.appCheck().token(forcingRefresh: forceRefresh, completion: { result, error in
            if let error = error {
                completion(nil, 0, error)
                return
            }
            var expireTimeMillis: Int = 0
            if let result = result {
                expireTimeMillis = Int(result.expirationDate.timeIntervalSince1970) * 1000
            }
            completion(result?.token, expireTimeMillis, nil)
        })
    }

    @objc public func initialize(debug: Bool) {
        if debug {
            AppCheck.setAppCheckProviderFactory(AppCheckDebugProviderFactory())
        } else {
            AppCheck.setAppCheckProviderFactory(CustomAppCheckProviderFactory())
        }
    }

    @objc public func setTokenAutoRefreshEnabled(_ enabled: Bool) {
        AppCheck.appCheck().isTokenAutoRefreshEnabled = enabled
    }
}

File: packages/app-check/android/src/main/java/io/capawesome/capacitorjs/plugins/firebase/appcheck/FirebaseAppCheck.java
package io.capawesome.capacitorjs.plugins.firebase.appcheck;

import android.util.Log;
import androidx.annotation.NonNull;
import com.google.firebase.appcheck.FirebaseAppCheck.AppCheckListener;
import com.google.firebase.appcheck.debug.DebugAppCheckProviderFactory;
import com.google.firebase.appcheck.playintegrity.PlayIntegrityAppCheckProviderFactory;

public class FirebaseAppCheck {

    @NonNull
    private final FirebaseAppCheckPlugin plugin;

    @NonNull
    private final AppCheckListener appCheckListener;

    public FirebaseAppCheck(@NonNull FirebaseAppCheckPlugin plugin) {
        this.plugin = plugin;
        this.appCheckListener = this.plugin::handleTokenChanged;
        getFirebaseAppCheckInstance().addAppCheckListener(appCheckListener);
    }

    public void getToken(boolean forceRefresh, final GetTokenResultCallback resultCallback) {
        getFirebaseAppCheckInstance()
            .getAppCheckToken(forceRefresh)
            .addOnSuccessListener(appCheckToken -> {
                resultCallback.success(appCheckToken.getToken(), appCheckToken.getExpireTimeMillis());
            })
            .addOnFailureListener(exception -> {
                Log.w(FirebaseAppCheckPlugin.TAG, "Get App Check token failed.", exception);
                resultCallback.error(exception.getMessage());
            });
    }

    public void initialize(boolean debug, boolean isTokenAutoRefreshEnabled) {
        if (debug) {
            getFirebaseAppCheckInstance()
                .installAppCheckProviderFactory(DebugAppCheckProviderFactory.getInstance(), isTokenAutoRefreshEnabled);
        } else {
            getFirebaseAppCheckInstance()
                .installAppCheckProviderFactory(PlayIntegrityAppCheckProviderFactory.getInstance(), isTokenAutoRefreshEnabled);
        }
    }

    public void setTokenAutoRefreshEnabled(boolean enabled) {
        getFirebaseAppCheckInstance().setTokenAutoRefreshEnabled(enabled);
    }

    private com.google.firebase.appcheck.FirebaseAppCheck getFirebaseAppCheckInstance() {
        return com.google.firebase.appcheck.FirebaseAppCheck.getInstance();
    }
}
